src/Uno.Extensions.Reactive.Tests/_Utils/NotifyCollectionChangedComparer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using Uno.Extensions.Collections;
+
+namespace Uno.Extensions.Reactive.Tests._Utils
+{
+	internal class NotifyCollectionChangedComparer : IComparer, IEqualityComparer, IEqualityComparer<NotifyCollectionChangedEventArgs>
+	{
+		private readonly IEqualityComparer<object> _itemComparer;
+		public static NotifyCollectionChangedComparer Default { get; } = new NotifyCollectionChangedComparer(EqualityComparer<object>.Default);
+
+		public NotifyCollectionChangedComparer(IEqualityComparer<object> itemComparer)
+		{
+			_itemComparer = itemComparer;
+		}
+
+		public int Compare(object? l, object? r)
+			=> Equals(l, r)
+				? 0
+				: -1;
+
+		public bool Equals(NotifyCollectionChangedEventArgs? left, NotifyCollectionChangedEventArgs? right)
+		{
+			if (left == null)
+			{
+				return right == null;
+			}
+
+			return right != null
+				&& left.Action == right.Action
+				&& left.OldStartingIndex == right.OldStartingIndex
+				&& AreSequenceEquals(left.OldItems, right.OldItems)
+				&& left.NewStartingIndex == right.NewStartingIndex
+				&& AreSequenceEquals(left.NewItems, right.NewItems)
+				&& (
+					!(left is RichNotifyCollectionChangedEventArgs richLeft)
+					|| !(right is RichNotifyCollectionChangedEventArgs richRight)
+					|| (
+						AreSequenceEquals(richLeft.ResetOldItems, richRight.ResetOldItems)
+						&& AreSequenceEquals(richLeft.ResetNewItems, richRight.ResetNewItems)
+					)
+				);
```

Comment:
Is the order by estimated order of possible exclusions?

Reply:
Yes as we validate the `Action` first, and no since there is no real order for other properties. Anyway it's only a test helper so perf is not really a concern here.